src/mpyl/cli/commands/projects/lint.py

```diff
@@ -91,19 +85,17 @@ def _assert_correct_project_linkup(
     console: Console,
     target: Target,
     projects: list[Project],
-    all_projects: list[Project],
     pr_identifier: Optional[int],
 ) -> list[WrongLinkupPerProject]:
     console.print("")
     console.print("Checking namespace substitution: ")
     wrong_substitutions = __get_wrong_substitutions_per_project(
-        all_projects, projects, pr_identifier, target
+        projects, pr_identifier, target
```

Comment:
I went through the exact same experience, _"this can't possibly be right"_ 😂 It is correct though:

This method is only called when we run `projects lint`, in [a single place in `projects.py`](https://github.com/Vandebron/gh-mpyl/pull/20/files/641929b294edfaf42f9a43edb2ce36717742279b#diff-ed7e8874848814ca1fa9b17a1166812bcff9bd772f0acfcf984552e6019cc16fL128-L142).
That method works roughly like this:
```python
def lint(
  filter: str = ""
):
  loaded_projects = load_projects(filter)

  if filter == "":
    all_projects = loaded_projects
  else:
    all_projects = load_projects("")
  
  _assert_correct_project_linkup(
    projects=loaded_projects,
    all_projects=all_projects,
  )
```
And since we never pass `filter` in any of our calls to MPyL, that else never kicks in and thus `all_projects` and `loaded_projects` are always the same.

Reply:
The reason of my confusion (and probably yours as well) is that eventually this lint check will call `def substitute_namespaces()`, which **does** get called twice:
* when we run `projects lint` ([always sending all projects in both parameters](https://github.com/Vandebron/gh-mpyl/blob/8244c594a04a1c2afaa2450b70752a1d531cb071/src/mpyl/cli/commands/projects/lint.py#L117-L122))
* when we run the `deploy` stage ([and there we **do** send different arguments](https://github.com/Vandebron/gh-mpyl/blob/main/src/mpyl/steps/deploy/k8s/chart.py#L749-L767): all projects vs projects in the run plan)

Reply:
Ahhh! Thanks for explaining